constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintSet.java

```diff
@@ -1106,6 +1107,48 @@ static void parseDimensionConstraintsString(Object data, String value, int orien
         }
     }
 
+
+    /**
+     * Get the types associated with this ConstraintSet
+     * The types mechanism allows you to tag the constraint set
+     * with a series of string to define properties of a ConstraintSet
+     *
+     * @return an array of type strings
+     */
+    public String[] getConstraintSetTypes() {
+        return Arrays.copyOf(mMatchTypes, mMatchTypes.length);
+    }
+
+    /**
+     * Set the types associated with this ConstraintSet
+     * The types mechanism allows you to tag the constraint set
+     * with a series of string to define properties of a ConstraintSet
+     * @param types a comer seprated array of strings.
+     */
+    public void setConstraintSetTypes(String types) {
```

Comment:
If this is just for XML parsing i'd say it should be package private and renamed.
```Java
void parseConstraintSetTypes(String types)
```

And add another function meant to be used programmatically
```Java
public void setConstraintSetTypes(String... types) {
```